src/lib.rs

```diff
@@ -150,7 +150,7 @@ pub fn run(
 
         if vm.predicate_holds(&opcode.predicate) {
             let result = match opcode.variant {
-                Variant::Invalid(_) => todo!(),
+                Variant::Invalid(_) => Err(EraVmError::OpcodeError(OpcodeError::InvalidOpCode)),
```

Comment:
nit: you could do `OpcodeError::InvalidOpCode.into()` since `EraVmError` implements from for it